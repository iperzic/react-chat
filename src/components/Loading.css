.Loading {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 10;
}

.Loading__backdrop {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: #d4dade;
    opacity: 0.6;
}

.Loading__container {
    margin: 50% auto 0;
    width: 70px;
    text-align: center;
}

.Loading__container > div {
    width: 20px;
    height: 20px;
    background-color: #3798d4;
    border: 1px solid #2e77a4;
    border-radius: 100%;
    display: inline-block;
    animation: bounce 1.4s infinite ease-in-out both;
}

.Loading__container .Loading__bounce1 {
    animation-delay: -0.32s;
}

.Loading__container .Loading__bounce2 {
    animation-delay: -0.16s;
}

@keyframes bounce {
    0%, 80%, 100% {
        transform: scale(0);
    }
    40% {
        transform: scale(1.0);
    }
}
