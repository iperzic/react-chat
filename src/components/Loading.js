import React from 'react';

import './Loading.css';

const Loading = () => (
  <div className="Loading">
    <div className="Loading__backdrop" />
    <div className="Loading__container">
      <div className="Loading__bounce1" />
      <div className="Loading__bounce2" />
      <div className="Loading__bounce3" />
    </div>
  </div>
);


export default Loading;
