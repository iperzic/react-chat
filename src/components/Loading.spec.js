import React from 'react';
import { shallow } from 'enzyme';

import Loading from './Loading';

describe('Loading', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Loading />);
  });

  it('should show loader', () => {
    expect(component.getElement()).toMatchSnapshot();
  });
});
