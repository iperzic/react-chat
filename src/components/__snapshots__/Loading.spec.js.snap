// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should show loader 1`] = `
<div
  className="Loading"
>
  <div
    className="Loading__backdrop"
  />
  <div
    className="Loading__container"
  >
    <div
      className="Loading__bounce1"
    />
    <div
      className="Loading__bounce2"
    />
    <div
      className="Loading__bounce3"
    />
  </div>
</div>
`;
